rag/llm/key_utils.py

```
@@ -31,4 +34,64 @@ def _normalize_replicate_key(key):
+        except (json.JSONDecodeError, TypeError):
+            logging.warning(
+                "Bedrock key is not valid JSON; expected a JSON object with 'auth_mode' and 'bedrock_region' (see conf/models/bedrock.json).",
+            )
+            raise ModelException(
+                "Bedrock requires a JSON key with at least 'auth_mode' and "
+                '\'bedrock_region\'. Example: {"auth_mode": "access_key_secret", '
+                '"bedrock_region": "us-east-1", "bedrock_ak": "...", '
+                '"bedrock_sk": "..."}. See conf/models/bedrock.json for the '
+                "full schema.",
+                retryable=False,
+            )
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)